server/src/main/java/org/elasticsearch/cluster/routing/allocation/DiskThresholdMonitor.java

```diff
@@ -155,6 +162,10 @@ public void onNewInfo(ClusterInfo info) {
                 continue;
             }
 
+            if (isFrozenOnlyNode(routingNode)) {
```

Comment:
Could you add a short comment here about why we skip frozen nodes?

Reply:
Added in ff5250f59aed0d961375d49cc761dc5093d8c61e